src/mono/wasm/host/CommonConfiguration.cs

```diff
@@ -125,7 +125,7 @@ public ProxyOptions ToProxyOptions()
     public static void CheckPathOrInAppPath(string appPath, string? path, string argName)
     {
         if (string.IsNullOrEmpty(path))
-            throw new ArgumentNullException($"Missing value for {argName}");
+            throw new ArgumentException($"Missing value for {argName}");
```

Comment:
Can this instead just be:
```C#
ArgumentException.ThrowIfNullOrEmpty(argName);
```
?

Reply:
I think the message does provide value.  It's indicating that there is a property missing in "runtimeconfig.json".

Reply:
```C#
string argName = "runtimeconfig.json";

try
{
    throw new ArgumentException($"Missing value for {argName}");
}
catch (Exception e)
{
    Console.WriteLine(e.Message);
}

try
{
    ArgumentException.ThrowIfNullOrEmpty("", argName);
}
catch (Exception e)
{
    Console.WriteLine(e.Message);
}
```
outputs:
```
Missing value for runtimeconfig.json
The value cannot be an empty string. (Parameter 'runtimeconfig.json')
```
The latter is pretty clear about the problem.

I don't have strong feelings about it. Just doesn't seem worth the extra code.

Reply:
I would prefer using `CommandLineException` instead of `Argument**` for all occurences in this method. We have a catch for `CommandLineException` and show the message without stacktrace etc https://github.com/dotnet/runtime/blob/e733539a38458c5c3a64688b55a702444e470700/src/mono/wasm/host/Program.cs#L52-L56